hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/states/endpoint/RegisterEndpointTask.java

```diff
@@ -163,7 +163,8 @@ public EndpointStateMachine.EndPointStates call() throws Exception {
         Preconditions.checkState(!StringUtils.isBlank(response.getClusterID()),
             "Invalid cluster ID in the response.");
         Preconditions.checkState(response.getErrorCode() == success,
-            "DataNode has higher Software Layout Version than SCM.");
+            "DataNode has different Software Layout Version" +
```

Comment:
Could you log the address (host / port) it is connecting to that failed the check? Is that information available in the rpcEndPoint object?

Reply:
@sodonnel Thanks for your review, I made some changes, can you take a look again